torchgeo/datasets/cloud_cover.py

```diff
@@ -0,0 +1,390 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Cloud Cover Detection Challenge dataset."""
+
+import json
+import os
+from typing import Any, Callable, Dict, List, Optional, Sequence
+
+import matplotlib.pyplot as plt
+import numpy as np
+import rasterio
+import torch
+from numpy.typing import NDArray
+from torch import Tensor
+
+from .geo import VisionDataset
+from .utils import check_integrity, download_radiant_mlhub_dataset, extract_archive
+
+
+# TODO: read geospatial information from stac.json files
+class CloudCoverDetection(VisionDataset):
+    """Cloud Cover Detection Challenge dataset.
+
+    This training dataset was generated as part of a
+    [crowdsourcing competition](https://www.drivendata.org/competitions/83/cloud-cover/)
+    on DrivenData.org, and later on was validated using a team of expert annotators. See
+    `this website<https://mlhub.earth/data/ref_cloud_cover_detection_challenge_v1>`
+    for dataset details.
+
+    The dataset consists of Sentinel-2 satellite imagery and corresponding cloudy
+    labels stored as GeoTiffs. There are 22,728 chips in the training data,
+    collected between 2018 and 2020.
+
+    Each chip has:
+
+    * 4 multi-spectral bands from Sentinel-2 L2A product. The four bands are
+    [B02, B03, B04, B08] (refer to Sentinel-2 documentation for more
+    information about the bands).
+    * Label raster for the corresponding source tile representing a binary classifcation
+    for if the pixel is is a cloud or not.
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://doi.org/10.34911/RDNT.HFQ6M7
+
+    .. note::
+
+       This dataset requires the following additional library to be installed:
+
+       * `radiant-mlhub <https://pypi.org/project/radiant-mlhub/>`_ to download the
+         imagery and labels from the Radiant Earth MLHub
```

Comment:
You should add a line like:
```rst
    .. versionadded:: 0.3                                                                
```
to the end of this docstring. This documents which version of TorchGeo added this dataset.

Just a heads up, we're planning a 0.3.0 release in the next few days. If this is finished after 0.3 is released, just change this to 0.4.